mcode/mcode-extraction-framework
# modifications for base-engine to work

# Summary
Changes to support the implementation of a base-engine (called BaseClient) in the ICARE-extraction-client repository.

## New behavior/Code changes
- Made the baseFHIRExtractor functionality flexible enough to support extending it into various FHIRResourceExtractor classes.
- Updated the baseFHIRExtractor get parameters and constructor params
- Creation of various FHIRResourceExtractor classes. 
- Updating the CSVExtractors get parameters and constructor params
- Updated FHIR utils to use R4 versioning
- Avoid storing request headers locally as state in extractors.
- Update tests for existing CSV extractors and BaseFHIR
- Create tests for FHIRResourceExtractors

# Testing guidance
- Ensure that tests pass
- Ensure quality of the code 
- Connect this to the ICAREExtractionClient repo and ensure that running CSV client still works 
- Ensure auth still works for the WebService Client
